man/systemd.exec.xml

```diff
@@ -194,6 +194,42 @@
         <xi:include href="system-only.xml" xpointer="singular"/></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>MountPaths=</varname></term>
+
+        <listitem><para>This setting is similar to <varname>RootImage=</varname> in that it mounts a file
+        system hierarchy from a block device node or loopback file, but the destination directory can be
+        specified as well as mount options. This option expects a whitespace separated list of mount
+        definitions. Each definition consists of a colon-separated triple of source path, destination directory
+        and optional option string. Each mount definition may be prefixed with <literal>-</literal>, in which
+        case it will be ignored when its source path does not exist. The source argument is a path to a block
+        device node or regular file. The device node or file system image file needs to contain a file system
+        without a partition table. Refer to
+        <citerefentry><refentrytitle>mount</refentrytitle><manvolnum>8</manvolnum></citerefentry> for valid
+        mount options. Any mounts created with this option are specific to the unit, and are not visible in
+        the host's mount table.</para>
```

Comment:
i think we really should use the dissect logic for this too, i.e. support disk images with partition tables. it's easy, we have all the infrastructure for that already. See the systemd-dissect tool's --mount switch to see how this is used. Or RootImage=.

Reply:
The use case for this was to use multiple mounts to construct the mount namespace with fine control. For example, one mount for root, others for `/var`, `/var/xyz`, `/home/xyz` etc. Maybe that does not conflict with the dissect logic.

But another use case since the automatic dependency to udevd was added for RootImages was to circumvent those automatic dependencies with a lower level option which does not imply all those logic. So I was considering to use this for udevd itself and of course this would not be possible if there was again a dependency.

Reply:
The dissection logic is happy with images lacking partition tables too, so your usecase would be covered too.

If the non-dependency usecase is important to you we could maybe add some single-char decorator to RootImage= and this new option that turns it off. i.e. RootImage=+/foobar/waldo.img would mean: this is extra low-level shit, don#t generate a dep for it.